src/full/Agda/Interaction/Highlighting/Generate.hs

```diff
@@ -165,7 +165,7 @@ generateAndPrintSyntaxInfo
   -> TCM ()
 generateAndPrintSyntaxInfo decl _ _ | null $ P.getRange decl = return ()
 generateAndPrintSyntaxInfo decl hlLevel updateState = do
-  file <- fromMaybe __IMPOSSIBLE__ <$> asksTC envCurrentPath
+  file <- getCurrentPath
 
```

Comment:
It's only used in a few places, so probably not too hard to track down. And except for in the deprecated interactive mode, I think this will always be a `Just` anyway... For now, I prefer the cleaner code, but happy to change this if an `__IMPOSSIBLE__` actually shows up!

Reply:
Fair enough!